src/Avalonia.Visuals/Vector.cs

```diff
@@ -87,24 +83,25 @@ public Vector(double x, double y)
         /// <param name="scale">The divisor.</param>
         /// <returns>The scaled vector.</returns>
         public static Vector operator /(Vector vector, double scale)
-        {
-            return new Vector(vector._x / scale, vector._y / scale);
-        }
+            => Divide(vector, scale);
 
         /// <summary>
         /// Length of the vector
         /// </summary>
         public double Length => Math.Sqrt(X * X + Y * Y);
 
+        /// <summary>
+        /// Squared Length of the vector
+        /// </summary>
+        public double SquaredLength => Math.Sqrt(Length);
```

Comment:
This should be `X * X + Y * Y`. `Length` can be written as `Math.Sqrt(SquaredLength)`.

Reply:
yep. Implemented both separately because 'Math.Sqrt(SquaredLength)' seems more unclear than 'Math.Sqrt(_x * _x + _y * _y)' to me.